### PR TITLE
feat: Integrate core crypto logger (WPB-14636)

### PR DESCRIPTION
**Pull request by thisisamir98**
## Description

Description:
This PR connects the core crypto logging mechanism to our existing log system. It uses the setLogger method to register a callback (coreCryptoLogger) that forwards log messages from core crypto to our logging framework.

Key changes:

Implemented a coreCryptoLogger to map core crypto log levels to our logger.
Used logFunctions to ensure proper handling of different log levels (DEBUG, INFO, WARN, ERROR, etc.).
Configured the log level to INFO using the setMaxLogLevel method.
This integration ensures that core crypto logs are forwarded to the client’s native log system with appropriate levels and context for easier debugging and monitoring.

### CODE REVIEW

**Review by aweiss-dev — APPROVED** [review]

**Review by V-Gira — APPROVED** [review]

File: packages/core/src/messagingProtocols/proteus/ProteusService/CryptoClient/CoreCryptoWrapper/CoreCryptoWrapper.ts [hunk 1]
```diff
@@ -63,6 +81,10 @@ export async function buildClient(
     key: Encoder.toBase64(key.key).asString,
     wasmFilePath,
   });
+
+  setLogger(coreCryptoLogger);
+  setMaxLogLevel(CoreCryptoLogLevel.Error);
```
Comment (typfel) [inline packages/core/src/messagingProtocols/proteus/ProteusService/CryptoClient/CoreCryptoWrapper/CoreCryptoWrapper.ts:86]: We would like the default log level to be `Info`.

File: packages/core/src/messagingProtocols/mls/MLSService/CoreCryptoMLSError.ts [hunk 2]
```diff
@@ -17,6 +17,21 @@
  *
  */
 
+const CORE_CRYPTO_ERROR_NAMES = {
+  ConversationAlreadyExists: 'ConversationAlreadyExists',
```
Comment (typfel) [inline packages/core/src/messagingProtocols/mls/MLSService/CoreCryptoMLSError.ts:21]: You need to include `MlsError` prefix:

MlsErrorConversationAlreadyExists

File: packages/core/src/messagingProtocols/mls/MLSService/CoreCryptoMLSError.ts [hunk 3]
```diff
@@ -34,17 +49,30 @@ export const CoreCryptoMLSError = {
 } as const;
 
 export const isCoreCryptoMLSWrongEpochError = (error: unknown): boolean => {
-  return error instanceof Error && error.message === CoreCryptoMLSError.DECRYPTION.WRONG_EPOCH;
+  return (
+    error instanceof Error &&
+    (error.message === CoreCryptoMLSError.DECRYPTION.WRONG_EPOCH || error.name === CORE_CRYPTO_ERROR_NAMES.WrongEpoch)
```
Comment (typfel) [inline packages/core/src/messagingProtocols/mls/MLSService/CoreCryptoMLSError.ts:54]: Why are you keep the old matching against the error message?

File: packages/core/src/messagingProtocols/mls/MLSService/CoreCryptoMLSError.ts (hunk 2 above)
Comment (thisisamir98) [reply packages/core/src/messagingProtocols/mls/MLSService/CoreCryptoMLSError.ts]: https://github.com/wireapp/wire-web-packages/pull/6749/commits/988f9907c370b1a2e4e3c392fdb0d9a5e0764c63

File: packages/core/src/messagingProtocols/mls/MLSService/CoreCryptoMLSError.ts [hunk 4]
```diff
@@ -17,43 +17,43 @@
...
+  DuplicateMessage: 'DuplicateMessage',
+  BufferedFutureMessage: 'BufferedFutureMessage',
+  MessageEpochTooOld: 'MessageEpochTooOld',
+  SelfCommitIgnored: 'SelfCommitIgnored',
+  UnmergedPendingGroup: 'UnmergedPendingGroup',
+  StaleProposal: 'StaleProposal',
+  StaleCommit: 'StaleCommit',
+  Other: 'Other',
+  SessionNotFound: 'SessionNotFound',
+  RemoteIdentityChanged: 'RemoteIdentityChanged',
```
Comment (typfel) [inline packages/core/src/messagingProtocols/mls/MLSService/CoreCryptoMLSError.ts:31]:
```suggestion
  SessionNotFound: 'ProteusErrorSessionNotFound',
  RemoteIdentityChanged: 'ProteusErrorRemoteIdentityChanged',
```

**Review by typfel — APPROVED** [review]

**Pull request merged**